fiwareskuld/phase/phase0c_change_category.py

```diff
@@ -38,59 +38,74 @@
+
+    def change_user_keystone(self, user_id):
+        """Change the user from trial to basic
+        We use this on our testing environment
+        :param user_id:
+        :return: nothing
+        """
+        path = '/domains/default/users/{0}/roles/{1}'
+        if q.get_type_fiware_user(user_id) != 'trial':
+            logging.error('{0} is not a trial user'.format(user_id))
+            return
+
+        keystone.role_assignments.client.put(path.format(user_id, basic))
+        keystone.role_assignments.client.delete(path.format(user_id, trial))
+
+    def change_user_via_idm(self, user_id, type):
+        """Change the user from trial to basic
+        This method must be used in production.
+        :param user_id:
+        :return: nothing
+        """
+        if q.get_type_fiware_user(user_id) != type:
+            logger.error('{0} is not a {1} user'.format(user_id, type))
+        body = {'user_id': user_id, 'role_id': basic, 'notify': True}
+        headers = {'X-Auth-Token': q.osclients.get_token()}
+
+        # NOTE(garcianavalon) sometimes in settings people use the end /
+        horizon_url = HORIZON_ENDPOINT
+        if horizon_url.endswith('/'):
+            horizon_url = horizon_url[:-1]
+
+        r = requests.post(horizon_url + '/account_category',
+                          json=body, headers=headers, verify=False)
+        if r.status_code not in (200, 204):
+            msg = 'The operation returned code {0}: {1}'
+            logger.error(msg.format(r.status_code, r.reason))
+            raise Exception('Server error')
+
+    def change_to_basic_for_trial_users(self):
+        users = open('trial_users_to_delete.txt')
+        self._change_to_basic(users, "trial")
+
+    def change_to_basic_for_community_users(self):
+        users = open('community_users_to_delete.txt')
+        self._change_to_basic(users, "community")
+
+    def _change_to_basic(self, users, type):
+        for line in users.readlines():
+            user_id = line.strip().split(',')[0]
+            if user_id == '':
+                continue
+            logger.info('Changing user {0} from {1} to basic'.format(user_id, type))
+            try:
+                self.change_user_via_idm(user_id, type)
+            except Exception, e:
+                logger.error('Error changing user {0} to basic. Cause: {1}'.format(
+                    user_id, str(e)))
+
 
 if __name__ == '__main__':
-    users = open('users_to_delete.txt')
-    for line in users.readlines():
-        user_id = line.strip().split(',')[0]
-        if user_id == '':
-            continue
-        logger.info('Changing user {0} from trial to basic'.format(user_id))
-        try:
-            change_user_via_idm(user_id)
-        except Exception, e:
-            logger.error('Error changing user {0} to basic. Cause: {1}'.format(
-                user_id, str(e)))
+    change_category = ChangeCategory()
+    change_category.change_to_basic_for_community_users()
```

Comment:
I think that we should differentiate when we want to change category for Trial and for Community. I mean, we will execute the scripts in different moment in time to delete or Trial or Community but usually not both at the same time.